### ISSUE TITLE
Internet radio and other

### ISSUE THREAD

**Issue by kaaholst** [kaaholst/android-squeezer#2]
Hi Nik,

Nice to see you joining this project.

I have fetched your master branch, and merged my commits into it.
It contains support for internet radio, and some other stuff, which I hadn't pushed to github. There are some layout improvements to better support different devices, and a fix for a possible NPE. I see that you have done some improvements on the layout, so I kept yours, where it conflicted. 

Once again thanks for your work, we are now using the new version, and it's working flawlessly.

Original issue reported on github.com/nikclayton/android-squeezer by kaaholst on 2011-08-02T21:19:24Z

**Comment by kaaholst** [comment]
Hi, thanks for the comments.

Right now I'm focusing on (in the cache-server-data branch) code that maintains a local cache of content from the server.

I'm doing this be implementing a ContentProvider (that's backed by a SQLite DB) that knows to go and fetch data from the server if it needs it, and a Cursor that can return default values for missing data, and can tell the ContentProvider to fetch data as necessary.

In theory this might simplify a lot of the code, because the native listview already supports showing data from a cursor bound to a content provider.  And it's a big speed up for anyone using the app who wants to scroll through (for example) their album list.

The album list code is almost complete -- I think I've got one bug to track down with the cache pre-fill dialog progress box.  Once that's done, I plan on doing another list (e.g., songs), and refactor the common code out in to a single subclass.  That should (hopefully) remove a lot of the code in com.danga.squeezer.itemlists.

Brad gave me admin rights to the Google Code project hosting, and I _think_ I can push a new build to Market as well.  Would you like to do that with your codebase?


Original issue reported on github.com/nikclayton/android-squeezer by nikclayton on 2011-08-23T19:31:11Z

**Comment by kaaholst** [comment]
Hi Nik,

I just got your message, but I'm off to Portugal for a short vacation early
in the morning, so I will have to get back to you next week.

Regards
Kurt

2011/8/23 nikclayton <
reply@reply.github.com>

> Hi, thanks for the comments.
> 
> Right now I'm focusing on (in the cache-server-data branch) code that
> maintains a local cache of content from the server.
> 
> I'm doing this be implementing a ContentProvider (that's backed by a SQLite
> DB) that knows to go and fetch data from the server if it needs it, and a
> Cursor that can return default values for missing data, and can tell the
> ContentProvider to fetch data as necessary.
> 
> In theory this might simplify a lot of the code, because the native
> listview already supports showing data from a cursor bound to a content
> provider.  And it's a big speed up for anyone using the app who wants to
> scroll through (for example) their album list.
> 
> The album list code is almost complete -- I think I've got one bug to track
> down with the cache pre-fill dialog progress box.  Once that's done, I plan
> on doing another list (e.g., songs), and refactor the common code out in to
> a single subclass.  That should (hopefully) remove a lot of the code in
> com.danga.squeezer.itemlists.
> 
> Brad gave me admin rights to the Google Code project hosting, and I _think_
> I can push a new build to Market as well.  Would you like to do that with
> your codebase?
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/nikclayton/android-squeezer/pull/1#issuecomment-1883170


Original issue reported on github.com/nikclayton/android-squeezer by kaaholst on 2011-08-23T20:05:43Z

**Comment by kaaholst** [comment]
Hi again,

It would be great if you publish the latest version. I have never done it,
so I can't be of much help, but i guess it involves making a signed release
build. Unless you are keen to try it out, you might want to wait a few days,
because I am currently implementing issue 11: Offer to enable WiFi when
connecting fails. By the way publishing the lastest version, will also fix
issue 2, 3, 7 and maybe 10 (I don't know about last.fm or squeezescrobbler).

I noticed the cache-server-data branch, it looks interesting. It is a
daunting task, as I am sure you know, but I saw you got off to a good start,
already solved the dirty cache problem by storing the last date the
squeezeserver scanned the music library.

Regards
Kurt Aaholst

2011/8/23 nikclayton <
reply@reply.github.com>

> Hi, thanks for the comments.
> 
> Right now I'm focusing on (in the cache-server-data branch) code that
> maintains a local cache of content from the server.
> 
> I'm doing this be implementing a ContentProvider (that's backed by a SQLite
> DB) that knows to go and fetch data from the server if it needs it, and a
> Cursor that can return default values for missing data, and can tell the
> ContentProvider to fetch data as necessary.
> 
> In theory this might simplify a lot of the code, because the native
> listview already supports showing data from a cursor bound to a content
> provider.  And it's a big speed up for anyone using the app who wants to
> scroll through (for example) their album list.
> 
> The album list code is almost complete -- I think I've got one bug to track
> down with the cache pre-fill dialog progress box.  Once that's done, I plan
> on doing another list (e.g., songs), and refactor the common code out in to
> a single subclass.  That should (hopefully) remove a lot of the code in
> com.danga.squeezer.itemlists.
> 
> Brad gave me admin rights to the Google Code project hosting, and I _think_
> I can push a new build to Market as well.  Would you like to do that with
> your codebase?
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/nikclayton/android-squeezer/pull/1#issuecomment-1883170


Original issue reported on github.com/nikclayton/android-squeezer by kaaholst on 2011-09-02T21:07:19Z

**Comment by kaaholst** [comment]
Hi -- I'm happy to wait until you've got the WiFi issue fixed.  Co-incidentally, that bit me earlier this week and I was going to think about a solution :-)

The cache-server-data branch opens other possibilities as well -- since the information's stored in a content provider it should be fairly easy to create home screen widgets that read from the content provider for example.


Original issue reported on github.com/nikclayton/android-squeezer by nikclayton on 2011-09-03T08:02:19Z

**Comment by kaaholst** [comment]
I've just commited changes for issue 11, enable WiFi. This doesn't make much sense, unless the the auto-connect setting was also implemented, so I did this. I am not sure everyone wants/need this behavior, so it can be enabled/disabled in the preferences. The emulator always report no WiFi so in the emulator is has to be disabled.
I see that issue 16: "No server config = walk the user through creating one" is in the same ballpark, so I will start doing that right away.
You are welcome to publish the current, or wait for issue 16.


Original issue reported on github.com/nikclayton/android-squeezer by kaaholst on 2011-09-05T21:24:23Z
